src/StackExchange.Exceptional.AspNetCore/ConfigSettings.cs

```diff
@@ -0,0 +1,172 @@
+using System.Runtime.Serialization;
+using System.Text.RegularExpressions;
+
+namespace StackExchange.Exceptional
+{
+    internal partial class ConfigSettings
+    {
+        public ConfigSettings(IConfiguration configuration)
+        {
+            configuration.Bind(this);
+        }
+
+        public string ApplicationName { get; set; }
+        public string DataIncludePattern { get; set; }
+
+        public ErrorStoreSettings ErrorStore { get; set; }
+        public class ErrorStoreSettings
+        {
+            public string Type { get; set; }
+
+            public string Path { get; set; }
+
+            public string ConnectionString { get; set; }
+
+            public string ConnectionStringName { get; set; }
+
+            public int? Size { get; set; }
+
+            public int? RollupSeconds { get; set; }
+
+            public int? BackupQueueSize { get; set; }
+
+            internal void Populate(Settings settings)
+            {
+                var storeSettings = settings.Store;
+                storeSettings.Type = Type ?? storeSettings.Type;
+                storeSettings.Path = Path ?? storeSettings.Path;
+                storeSettings.ConnectionString = ConnectionString ?? storeSettings.ConnectionString;
+#if !NETSTANDARD2_0
+            storeSettings.ConnectionStringName = ConnectionStringName ?? storeSettings.ConnectionStringName;
+#endif
+                storeSettings.Size = Size ?? storeSettings.Size;
+                if (RollupSeconds != null)
+                    storeSettings.RollupPeriod = TimeSpan.FromSeconds(RollupSeconds.Value);
+                storeSettings.BackupQueueSize = BackupQueueSize ?? storeSettings.BackupQueueSize;
+            }
+        }
+
+        public IgnoreSettings IgnoreErrors { get; set; }
+        public class IgnoreSettings
+        {
+            public List<string> Regexes { get; set; }
+
+            public List<string> Types { get; set; }
+
+            internal void Populate(Settings settings)
+            {
+                var ignoreSettings = settings.Ignore;
+
+                if (Regexes != null)
+                {
+                    foreach (var regex in Regexes)
+                    {
+                        if (regex.HasValue())
+                        {
+                            ignoreSettings.Regexes.Add(new Regex(regex, RegexOptions.IgnoreCase | RegexOptions.Singleline));
+                        }
+                    }
+                }
+
+                if (Types != null)
+                {
+                    foreach (var type in Types)
+                    {
+                        ignoreSettings.Types.Add(type);
+                    }
+                }
+            }
+        }
+
+        public LogFilterSettings LogFilters { get; set; }
+        public class LogFilterSettings
+        {
+            public List<LogFilter> Form { get; set; } = new List<LogFilter>();
+
+            public List<LogFilter> Cookies { get; set; } = new List<LogFilter>();
```

Comment:
These can just be `Dictionary<string, string>`, the reason the non-Core version isn't is *purely* because of backwards compatibility with existing configs...we can be cleaner here and eliminate the `LogFilter` class :)